### PR TITLE
added editor spacing

### PR DESCRIPTION
**Pull request by toger5**
MERGE AFTER #11207

(I again cherry picked a commit from another pr... (`fixed margin beeing squared`) if that does not work I need to drop that commit first anyways...) Removed that commit... since it isn't working.
so ready to merge now.

optionally adds spacing to widgets so that the editor looks cleaner (but there is less space. thats why it is an option...)
screen shot 2017-09-13 at 02 45 50
screen shot 2017-09-13 at 02 45 44
screen shot 2017-09-13 at 02 45 34

### CODE REVIEW

**Comment by akien-mga** [comment]
Needs a rebase now. I think you could also merge the two commits together. "added configurable editor spacing" :)

**Comment by toger5** [comment]
Sure

**Comment by djrm** [comment]
@toger5 these kind of chages are atomic, meaning you can have one commit without the other, or at least does not make sense, maybe a single commit would be best

**Comment by toger5** [comment]
Definitly. No pc accessible atm...

**Comment by toger5** [comment]
@akien-mga @djrm updated and squashed

File: editor/editor_themes.cpp [hunk 1]
```diff
@@ -332,11 +332,14 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	style_default->set_draw_center(true);
 
 	// Button and widgets
-	Ref<StyleBoxFlat> style_widget = style_default->duplicate();
+	const float extra_spacing = EDITOR_DEF("interface/theme/additional_spacing", 0.0);
 
+	Ref<StyleBoxFlat> style_widget = style_default->duplicate();
+	style_widget->set_default_margin(MARGIN_LEFT, (extra_spacing + default_margin_size + 6) * EDSCALE);
```
Comment (djrm) [inline editor/editor_themes.cpp]: +default_margin_size? why, also these kind of changes make the editor_themes more error prone, it is easy to forget to add extra spacing when doing changes. in any case add extra_spacing to default_margin_size, but again so it scales everything consistently

Comment (toger5) [reply editor/editor_themes.cpp]: yea this was a mistake. as you see just fixed it ;)

**Comment by toger5** [comment]
screen shot 2017-09-19 at 00 39 14

this is the current state.

File: editor/editor_themes.cpp (hunk 1 above)
Comment (toger5) [reply editor/editor_themes.cpp]: don't know how that got there. I did some weird git stuff for rebasing on master which. wanted to grab the changes above another branch... probably I broke it there...

**Pull request merged**